### DISCUSSION TITLE
Named group

### DISCUSSION THREAD

**Discussion by JesusTheHun** [tailwindlabs/tailwindcss#5182] [Ideas]
What [this plugin](https://github.com/ErickTamayo/tailwindcss-named-groups) does, basically, can be handled gracefully by the JIT engine. Here is an image that tells everything :

![Demo](https://github.com/ErickTamayo/tailwindcss-named-groups/blob/main/images/in-action.gif?raw=true)

```html
<div class="group-foo bg-white hover:bg-blue-500 ...">
  <p class="text-gray-900 group-foo-hover:text-white ...">
    New Project
  </p>
  <div class="group-bar bg-gray-100 hover:bg-green-500 ...">
    <p class="text-gray-500 group-bar-hover:text-white ...">
      Create a new project from a variety of starting templates.
    </p>
  </div>
</div>
```

Both the image and the markup comes from the plugin's repo.
The plugin does not currently handle JIT mode.

So currently JIT mode breaks my code :(

**Comment by ankundinly** [comment]
Nearly all modern regular expression engines support numbered capturing groups and numbered backreferences. Long regular expressions with lots of groups and backreferences may be hard to read.   >>   [jcpassociates](https://www.jcpenneykiosk.run/)

**Reply by JesusTheHun** [reply]
I don't think the JIT engine uses regular expression, because I have read numerous times that spacing was an issue to implement some feature request. I don't have all the intel, obviously, but I think space wouldn't be an issue with a regular expression. However it would be very difference performance wise, I think.

**Comment by dcastil** [comment]
I would like such a feature, too! Here is an example where I'd need this:

It's a table cell which shows a button when it's hovered. Also when you hover close to the edge of the cell, a resize indicator shows up. Both the button and the resize indicator need to get some styles when a parent element (a `group`) gets hovered, but it's two different groups where one is the parent of another, making it impossible to use the same `group` class on both.

https://user-images.githubusercontent.com/31006608/129772216-494e69fb-6d23-488f-bcff-8283fa276da6.mp4

I think with the JIT compiler it should be feasible to add `group-inner`, `group-inner-inner` or something similar to Tailwind.

**Comment by reinink** [comment]
Hey! Good news — we just added nested group and multiple peer support using variant modifiers in Tailwind CSS v3.2 🥳

You can read about it in our [announcement blog post](https://tailwindcss.com/blog/tailwindcss-v3-2#nested-group-and-multiple-peer-support-using-variant-modifiers), but for quick reference here’s what it looks like:

```html
<div class="group/sidebar ...">
  <!-- ... -->
  <div class="group/navitem ...">
    <a
      href="#"
      class="opacity-50 group-hover/sidebar:opacity-75 group-hover/navitem:bg-black/75"
    >
      <!-- ... -->
    </a>
  </div>
  <!-- ... -->
</div>
```

This lets you give each group a clear name that makes sense for that context on the fly, and Tailwind will generate the necessary CSS to make it work.
